atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -695,29 +704,92 @@ private void cellsCleanedUp() {
                         line("rows.add(", StreamMetadataRow, ".BYTES_HYDRATOR.hydrateFromBytes(cell.getRowName()));");
                     } line("}");
                     line(StreamIndexTable, " indexTable = tables.get", StreamIndexTable, "(t);");
-                    line("Set<", StreamIndexRow, "> indexRows = rows.stream()");
+                    line("executeUnreferencedStreamDiagnostics(indexTable, rows);");
+                    line("Map<", StreamMetadataRow, ", StreamMetadata> currentMetadata = metaTable.getMetadatas(rows);");
+                    line("Set<", StreamId, "> toDelete = Sets.newHashSet();");
+                    line("for (Map.Entry<", StreamMetadataRow, ", StreamMetadata> e : currentMetadata.entrySet()) {"); {
+                        line("if (e.getValue().getStatus() != Status.STORED) {"); {
```

Comment:
It looks like this would never delete unreferenced-but-stored streams. I don't think we could use this in the internal shopping product without the feature flag we discussed.